.github/workflows/typos.yml

```diff
@@ -0,0 +1,19 @@
+name: Check Typos
+on:
+  push:
+    branches:
+      - master
```

Comment:
That's exactly what this does one line below. Your suggestion will make it run twice on a PR. Once for the PR and another since it is a commit (to a branch different from `master`)